peterrichards-lr/liferay-portal
# LPS-164567 - Migrate `document.cookie` to new cookie API with user consent

Hi @peterrichards-lr, sorry if this is not meant for you. You're receiving this PR because you're the one who created the affected file. If you know who maintains the `site-initializer-liferaybotics` project, please let us know so we can send it to the right team. Thanks!

### References

- [LPS-164567](https://issues.liferay.com/browse/LPS-164567)

### Goal

Replace usage of `document.cookie` with new cookie API in `liferaybotics` site initializer.

### Description

In Frontend Infrastructure we recently added a new cookie management JS API that takes user consent into account when setting or reading cookie values. This API replaces the native `document.cookie` object, and it will soon be enforced by the source formatter.

To help make things easier, we are sending individual PRs to all teams that use `document.cookie`, replacing the usages with the new `Liferay.Util.Cookie` methods.

But this is just a first step. Since each use case is different, each product team must add extra changes, handling the case where the user hasn't consented however they see fit. You could display an alert, disable specific functionality, maybe even ignore it. It's up to your product team.

For this purpose, we have updated your `setCookie` method to return a boolean representing whether the cookie has been set or not.

I have used [this Confluence document](https://liferay.atlassian.net/wiki/spaces/ENGR/pages/1998849368/Liferay+cookies) to check the consent type of your cookies, but since they aren't listed, I have defaulted to `NECESSARY` consent type. This type is always enabled, and if it's the right one you won't need extra changes. 

### How to test this

You should check your component works in the following situations:

* With feature flag disabled
* With feature flag enabled
   * With cookie banner disabled
   * With cookie banner enabled
      * In explicit consent mode
      * In implicit consent mode
      
You can follow these steps to test all of these cases:
* Enable the feature flag by adding `feature.flag.LPS-142518=true` to `portal-ext.properties` and restarting portal
* Enable the cookie banner by navigating to `Control Panel > System Settings > Cookies`, checking the corresponding checkbox and clicking `Save`
* Enable implicit consent mode by unchecking the second checkbox in `Control Panel > System Settings > Cookies`
